### PR TITLE
Optimize new tracing further

### PR DESCRIPTION
**Pull request by jutaro**
This pull request addresses the remaining performance penalty observed after the optimization of the new tracing mechanism introduced in pull request #5272. The goal is to further optimize the code to minimize the performance impact associated with the new tracing features.

Changes Made
Implemented additional optimizations to mitigate the performance penalty against old tracing.
The optimization efforts aim to balance the introduction of new features with the need for improved performance.

### CODE REVIEW

File: trace-dispatcher/src/Cardano/Logging/Trace.hs [hunk 1]
```diff
@@ -248,6 +268,7 @@ withDetails (Trace tr) = Trace $
         (lc, Left e) -> (lc, Left e))
       tr
   where
+    {-# INLINE process #-}
```
Comment (mgmeier) [inline trace-dispatcher/src/Cardano/Logging/Trace.hs:271]: Only top-level definitions can be marked `INLINE`. As it's ignored here, can we remove it?

File: trace-dispatcher/src/Cardano/Logging/Tracer/Composed.hs [hunk 2]
```diff
@@ -87,24 +88,23 @@ mkCardanoTracer' trStdout trForward mbTrEkg tracerPrefix hook = do
...
-                        pure (metricsFormatter "" ekgTrace)
+                        pure (metricsFormatter ekgTrace)
 --                      >>= recordMetricsStatistics internalTr
                         >>= maybeSilent hasNoMetrics tracerPrefix True
                         >>= hook
 
     pure (messageTrace <> metricsTrace)
 
   where
+    {-# INLINE addContextAndFilter #-}
```
Comment (mgmeier) [inline trace-dispatcher/src/Cardano/Logging/Tracer/Composed.hs:99]: Same as above, `INLINE` works on top-level definitions only.

File: trace-dispatcher/src/Cardano/Logging/Types.hs [hunk 3]
```diff
@@ -53,7 +53,7 @@ module Cardano.Logging.Types (
 
 
 import           Codec.Serialise (Serialise (..))
-import           Data.Aeson ((.=))
+-- import           Data.Aeson ((.=))
```
Comment (mgmeier) [inline trace-dispatcher/src/Cardano/Logging/Types.hs:56]: Please remove commented out line (+ the ones below).

File: trace-dispatcher/src/Cardano/Logging/Trace.hs (hunk 1 above)
Comment (jutaro) [reply trace-dispatcher/src/Cardano/Logging/Trace.hs:271]: Done

File: trace-dispatcher/src/Cardano/Logging/Tracer/Composed.hs (hunk 2 above)
Comment (jutaro) [reply trace-dispatcher/src/Cardano/Logging/Tracer/Composed.hs:99]: Done

File: trace-dispatcher/src/Cardano/Logging/Types.hs (hunk 3 above)
Comment (jutaro) [reply trace-dispatcher/src/Cardano/Logging/Types.hs]: Done

**Review by mgmeier — APPROVED** [review]
Looks great, thank you @jutaro

**Pull request merged**